src/node/rpc/member_frontend.h

```diff
@@ -647,8 +672,41 @@ namespace ccf
         return make_success(
           Propose::Out{complete_proposal(args.tx, proposal_id, proposal)});
       };
-      make_endpoint("propose", HTTP_POST, json_adapter(propose))
+      make_endpoint("proposal", HTTP_POST, json_adapter(propose))
         .set_auto_schema<Propose>()
+        .set_require_client_signature(true)
+        .install();
+
+      auto get_proposal =
+        [this](EndpointContext& args, nlohmann::json&& params) {
+          if (!check_member_active(args.tx, args.caller_id))
+          {
+            return make_error(HTTP_STATUS_FORBIDDEN, "Member is not active");
+          }
+
+          ObjectId proposal_id;
+          std::string error;
+          if (!get_proposal_id(
+                args.rpc_ctx->get_request_path_params(), proposal_id, error))
+          {
+            return make_error(HTTP_STATUS_BAD_REQUEST, error);
+          }
+
+          auto proposals = args.tx.get_view(this->network.proposals);
+          auto proposal = proposals->get(proposal_id);
+
+          if (!proposal)
+          {
+            return make_error(
+              HTTP_STATUS_BAD_REQUEST,
+              fmt::format("Proposal {} does not exist", proposal_id));
+          }
+
+          return make_success(proposal.value());
+        };
+      make_endpoint("proposal/{id}", HTTP_GET, json_adapter(get_proposal))
+        .set_auto_schema<void, Proposal>()
+        //.set_require_client_signature(true)
```

Comment:
It seems odd to require a signature for a `GET` (turning it into a write!), but on the other hand it may be simpler to require signatures for _everything on the member frontend_, or at least _every operation on proposals_? I'm leaning towards saying this is unsigned and truly read-only.

Reply:
Leaning towards unsigned and truly read-only too as the proposal could be read from the ledger directly anyway.

Reply:
I was going to suggest that a PBFT world, where all requests including reads are persisted, it's nice to have the same auditability on governance reads that we have on writes? On the other hand, governance state is public in the ledger, so it can absolutely be read in a non-auditable way. So I agree with pure read-only without signature (another nail in the frontend coffin).

Reply:
I've made this truly read-only.